src/lib/gatk.ml

```diff
@@ -93,67 +93,149 @@ module Configuration = struct
+  reference_build:string ->
+  processors:int ->
+  run_with:Run_environment.Machine.t ->
+  a KEDSL.bam_or_bams ->
+  a =
+  fun ?(compress=false)
+    ~configuration ~reference_build ~processors ~run_with input_bam_or_bams ->
+    let open KEDSL in
+    let input_bam_1, more_input_bams = (* this an at-least-length-1 list :)  *)
+      match input_bam_or_bams with
+      | Single_bam bam -> bam, []
+      | Bam_workflow_list [] -> 
+        failwithf "Empty bam-list in Gatk.indel_realigner`"
+      | Bam_workflow_list (one :: more) -> (one, more)
+    in
+    let indel_config, target_config = configuration in
+    let input_sorted_bam_1 =
+      Samtools.sort_bam_if_necessary
+        ~run_with ~processors ~by:`Coordinate input_bam_1 in
+    let more_input_sorted_bams =
+      List.map more_input_bams
+        ~f:(Samtools.sort_bam_if_necessary
+              ~run_with ~processors ~by:`Coordinate) in
+    let name = 
+      sprintf "gatk-indelrealign-%dx-%s" 
+        (List.length more_input_bams + 1)
+        (Filename.basename input_sorted_bam_1#product#path) in
+    let gatk = Machine.get_tool run_with Tool.Default.gatk in
+    let reference_genome =
+      Machine.get_reference_genome run_with reference_build in
+    let fasta = Reference_genome.fasta reference_genome in
+    let digest_of_input =
+      List.map (input_sorted_bam_1 :: more_input_sorted_bams)
+        ~f:(fun o -> o#product#path)
+      |> String.concat ~sep:"" 
+      (* we make this file “unique” with an MD5 sum of the input paths *)
+      |> Digest.string |> Digest.to_hex in
+    let output_suffix =
+      sprintf "-target-%s-%dx-%s"
+        target_config.Configuration.Indel_realigner.name
+        (List.length more_input_bams + 1)
+        (if more_input_bams = [] then "" else "-" ^ digest_of_input)
+    in
+    let intervals_file =
+      Filename.chop_suffix input_sorted_bam_1#product#path ".bam" 
+      ^ output_suffix ^ ".intervals" in
+    let make =
+      let target_creation_args =
+        [
+          "-R"; Filename.quote fasta#product#path;
+          "-I"; Filename.quote input_sorted_bam_1#product#path;
+          "-o"; Filename.quote intervals_file;
+          "-nt"; Int.to_string processors
+        ]
+        @ Realigner_target_creator.render target_config
+        @ List.concat_map more_input_bams
+          ~f:(fun bam -> ["-I"; Filename.quote bam#product#path])
+      in
+      let indel_real_args =
+        [ "-R"; fasta#product#path;
+          "-I"; input_sorted_bam_1#product#path;
+          "-targetIntervals"; intervals_file;
+        ] @ Indel_realigner.render indel_config @
+        begin match more_input_bams with
+        | [] ->
+          ["-o";
+           Filename.chop_extension
+             input_sorted_bam_1#product#path ^ output_suffix ^ ".bam" ;]
+        | more ->
+          List.concat_map more
+            ~f:(fun b -> ["-I"; Filename.quote b#product#path])
+          @ ["--nWayOut"; output_suffix ^ ".bam"]
```

Comment:
Think we need to escape the `output_suffix` or at least not start with `-`:

```
##### ERROR ------------------------------------------------------------------------------------------
##### ERROR A USER ERROR has occurred (version 3.5-0-g36282e4): 
##### ERROR
##### ERROR This means that one or more arguments or inputs in your command are incorrect.
##### ERROR The error message below tells you what is the problem.
##### ERROR
##### ERROR If the problem is an invalid argument, please check the online documentation guide
##### ERROR (or rerun your command with --help) to view allowable command-line arguments for this tool.
##### ERROR
##### ERROR Visit our website and forum for extensive documentation and answers to 
##### ERROR commonly asked questions http://www.broadinstitute.org/gatk
##### ERROR
##### ERROR Please do NOT post this error to the GATK forum unless you have really tried to fix it yourself.
##### ERROR
##### ERROR MESSAGE: Argument with name 'target-ignore-mismatch-2x--b7beee7a8d8b58aac8b3d9321229954e.bam' isn't defined.
##### ERROR ------------------------------------------------------------------------------------------
```

But the call was actually `` --nWayOut -target-ignore-mismatch-2x--b7beee7a8d8b58aac8b3d9321229954e.bam`` (with one `-` that's not in the error message)